### PR TITLE
Backport clang resource-header fixes from next

### PR DESCRIPTION
**Pull request by etcwilde**
Cherry-picking changes from next that remove the files that get installed by multiple targets.
This fixes the failing Xcode generator configuration failure for Swift.

### CODE REVIEW

**Comment by etcwilde** [comment]
@swift-ci please test

File: clang/lib/Headers/CMakeLists.txt [hunk 1]
```diff
@@ -543,7 +605,17 @@ if (NOT LLVM_ENABLE_IDE)
   add_llvm_install_targets(install-webassembly-resource-headers
                            DEPENDS webassembly-resource-headers
                            COMPONENT webassembly-resource-headers)
+
+    add_llvm_install_targets(install-opencl-resource-headers
```
Comment (compnerd) [inline clang/lib/Headers/CMakeLists.txt]: Indentation is incorrect.

**Review by compnerd — COMMENTED** [review]
I think that I am tempted to say that this is better done as cherry-picking the revert and re-application instead.  I don't see anything egregiously incorrect in the application as done, but the serial cherry-picks makes it easier to associate the change with the upstream changes.

```
git cherry-pick -x 38822b98fa3b9d740b9a68b0de34296205d20819 a70cf56d20b956fc008d24420e14226127ba9f4a
```
would also make it easier to associate this this with the reported issue upstream at llvm/llvm-project#55002.

**Comment by etcwilde** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]
Thanks!

**Pull request merged**